tests/commands/test_bump_command.py

```diff
@@ -1449,3 +1449,15 @@ def test_bump_changelog_contains_increment_only(mocker, tmp_commitizen_project,
 
     assert "3.0.0" in out
     assert "2.0.0" not in out
+
+
+def test_bump_command_shows_description_when_use_help_option(
+    mocker: MockFixture, capsys
+):
+    testargs = ["cz", "bump", "--help"]
+    mocker.patch.object(sys, "argv", testargs)
+    with pytest.raises(SystemExit):
+        cli.main()
+
+    out, _ = capsys.readouterr()
+    assert "bump semantic version based on the git log" in str(out).replace("\n", " ")
```

Comment:
none blocker, but we could try something like https://github.com/commitizen-tools/commitizen/blob/f1ece99dc9ca142268b912671b04dfdfe5addb3f/tests/test_bump_update_version_in_files.py#L157-L158 as well

Reply:
I don't know if I understand... I wrote the `test_bump_command_shows_description_when_use_help_option` to assert that the command description shows on running `--help` against the command. Can you explain your idea with other words?

Reply:
`pytest-regeression` is a tool that you can verify whether the content (e.g., str, data frame) is exactly the same as the last time we generated it. It done so by creating a text file that records the last result. https://pytest-regressions.readthedocs.io/en/latest/overview.html

Reply:
I will look to this later, today. Thanks!

Reply:
@Lee-W `pytest-regressions` is a game changer! Using the assert only approach isn't enough to check if the command descriptions matches. For example:

Using `assert "create new commit" in str(out).replace("\n", " ")` isn't reliable because if we add an dot to the command description "create new commit." it still matching.

But, using `file_regression` we can check for the entire output and adding the dot to the end of the command description will cause the test to fail.

I replaced all the assertion approach for the `file_regression`, except for two: **changelog** and **bump**.

Analyzing the failure I found the "problem" is in the file `commitizen.version_schemes.py`. Theres an constant called **KNOWN_SCHEMES** that is an set of all available scheme providers. The problem is: set is an unordered struct so the order changes for every call. Since the order changes for every call, we cannot assure using `file_regression`:

![image](https://github.com/commitizen-tools/commitizen/assets/48416792/be8af65c-2888-4203-b835-600a9ec38421)

To fix this we can just replace the set comprehension for an list comprehension:
```diff
- KNOWN_SCHEMES = {ep.name for ep in metadata.entry_points(group=SCHEMES_ENTRYPOINT)}
+ KNOWN_SCHEMES = [ep.name for ep in metadata.entry_points(group=SCHEMES_ENTRYPOINT)]
```

I change the set for list and if passes on all tests, and now we can use `file_regression`. Theres an reason this to be an set? We can change to list?

Reply:
I'm working on the changes requested on #1115 (move step from generate_cli_screenshots.yml to docspublish.yml) and theres an another problem caused by using the set comprehension:

![image](https://github.com/commitizen-tools/commitizen/assets/48416792/4a7b788e-edb0-465c-8078-df25353a983e)

Even when thats no real changes, **bump** and **changelog** svg are being re-generated due to the order change on **KNOWN_SCHEMES**

Reply:
> I change the set for list and it passes on all tests, and now we can use file_regression. Theres an reason this to be an set? We can change to list?

I guess that's for performance and the fact that it didn't really need to be a list. But it could be a list in this case. Nice catching!

Reply:
@marcosdotme Have we pushed the `file_regression` change?

Reply:
Waiting more opinions about that, so I can create an issue, resolve then, merge into the branch of this PR an it will automatically pass all tests.

Reply:
@Lee-W Not yet! Is in stage. I was waiting for any thoughts about the **KNOWN_SCHEMES** "problem"

Reply:
I think we're good to change `KNOWN_SCHEMES` to list and change test cases to file_regressions. Let me know if I missed anything 🙂

Reply:
Just modify this one should be fine 🙂

Reply:
@Lee-W  I will wait to #1126 be merged because this branch will also fail on CI due to trailing whitespaces on ***.svg**

Reply:
It's merged 🎉

Reply:
@Lee-W After run the `./scripts/test` it fails because the missing ":" on commit message:

![image](https://github.com/commitizen-tools/commitizen/assets/48416792/500eaff7-d6e0-4531-bfa1-b957c1d931fd)

The commit is:

![image](https://github.com/commitizen-tools/commitizen/assets/48416792/40b496aa-8ab6-4630-9bdf-d8a11e3bedba)

How to proceed? 🥲

Reply:
oh just ignore it